### ISSUE TITLE
Crash alert when quitting IINA 1.3.1 while on fullscreen

### ISSUE THREAD

**Issue by byGits** [iina/iina#4082]
**System and IINA version:**

- macOS 12.5.1
- IINA 1.3.1

**Expected behavior:**

Pressing Command+Q should quit the application in any state without an error.

**Actual behavior:**

Every time I quit IINA 1.3.1 using Command+Q while the video is fullscreen I'm getting a crash alert (IINA quit unexpectedly.)
This was not the case with IINA 1.3.0 as far as I remember.

<details>
<summary><b>Crash report:</b></summary>

```
Crashed Thread:        0

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000048
Exception Codes:       0x0000000000000001, 0x0000000000000048
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [3637]

VM Region Info: 0x48 is not in any region.  Bytes before following region: 105553518919608
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      MALLOC_NANO (reserved)   600018000000-600020000000 [128.0M] rw-/rwx SM=NUL  ...(unallocated)
```
</details>

<br>

<details>
<summary><b>mpv log:</b></summary>

```
```
</details>

<br>

**Steps to reproduce:**

Open any video > Fullscreen > Command+Q

- [ ] MPV does not have this problem.

**How often does this happen?**

Every time while in fullscreen.

**Comment by contepaccio** [comment]
Same problem for me!

macOS 10.13.6
IINA 1.3.1

**Comment by low-batt** [comment]
Thanks for the report!

The crash report is truncated, but I'm sure from the description this is a duplicate of #4020 (and #4045). A fix is waiting to be reviewed in PR #4068. As we already have two open issues I will be closing this one as a duplicate.

The potential for this crash has been in the code for a long time. Whether IINA encountered this crash depended upon timing (thread race condition). Previously the timing was such that this crash normally did not happen. Fixes in 1.3.1 for other issues and crashes during quitting changed the timing and made it much more likely this problem would occur. The fix adds proper thread coordination to eliminate the race condition.

**Comment by low-batt** [comment]
Closed as duplicate of #4020.
